File: androidApp/src/main/java/com/luisma/conexiones/android/features/game/GameViewEvents.kt
package com.luisma.conexiones.android.features.game

sealed class GameViewEvents {
    data class OnCreate(val gameId: Int) : GameViewEvents()
    data class Refresh(val gameId: Int) : GameViewEvents()
    data class SelectWord(val col: Int, val row: Int) : GameViewEvents()
    data class DismissWordAnimation(val col: Int, val row: Int) : GameViewEvents()
    data class DismissTileAnimation(val row: Int) : GameViewEvents()
    data class GoToNextLevel(val nextLevelId: Int) : GameViewEvents()
    data object EnableLivesButton : GameViewEvents()
    data object OnAddWatched: GameViewEvents()
    data object MixGame : GameViewEvents()
    data object ClearSelection : GameViewEvents()
    data object Submit : GameViewEvents()
    data object OnBack : GameViewEvents()
    data object DismissLivesAnimation : GameViewEvents()
    data object DismissDoneSignAnimation : GameViewEvents()
}

File: iosApp/iosApp/Features/Game/GameViewEvents.swift
//
//  GameViewEvents.swift
//  iosApp
//
//  Created by Luis Manuel García Fernández on 24/10/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import SwiftUI
import shared

enum GameViewEvents {
    case onAppear(gameId: Int)
    case selectWord(column: Int, row: Int)
    case dismissAnimation(column: Int, row: Int)
    case goToNextLevel(nextLevel: Int)
    case dismissTileAnimation(row: Int)
    case dismissOnDoneAnimation
    case dismissLivesAnimation
    case mixGame
    case clearSelection
    case onBack
    case submit
    case moreLives
}
